app/src/main/java/lt/getpet/getpet/fragments/UserProfileFragment.kt

```diff
@@ -73,7 +80,23 @@ class UserProfileFragment : BaseFragment(), AuthStateChangedListener {
         }
     }
 
+    private fun onUserSignedOut(context: Context) {
+        subscriptions.clear()
+        val disposable = authenticationManager.signOutUser(context)
+                .subscribeOn(ioScheduler)
+                .observeOn(uiScheduler)
+                .subscribe({
+                    showAnonymousProfile()
+                }, {
+                    Timber.w(it)
+                    Toast.makeText(context, it.localizedMessage, Toast.LENGTH_LONG).show()
+                })
+
+        subscriptions.add(disposable)
+    }
+
     override fun onUserLoggedIn(user: User) {
+        subscriptions.clear()
```

Comment:
I would recommend to avoid clearing subscriptions in every place. 
It will be quite difficult to track down problem if we add one more `disposable` not related to logout into `subscriptions bag` and it would disappear.